tools/crm_shadow.c

```diff
@@ -878,17 +894,18 @@ show_shadow_contents(GError **error)
     filename = get_shadow_file(options.instance);
 
     if (check_file_exists(filename, true, error) == pcmk_rc_ok) {
-        char *output_s = NULL;
         xmlNode *output = NULL;
+        bool quiet_orig = out->quiet;
 
         if (read_xml(filename, &output, error) != pcmk_rc_ok) {
             goto done;
         }
 
-        output_s = dump_xml_formatted(output);
-        printf("%s", output_s);
+        out->quiet = true;
+        out->message(out, "shadow",
+                     options.instance, NULL, output, NULL, shadow_disp_content);
+        out->quiet = quiet_orig;
```

Comment:
Do we need to save and reset quiet, or can we just set it once in main()?

Rather than forcing it, we could set quiet only if no verbose options were given. Then you could test your non-quiet text output.

Reply:
It's kind of a pain either way :) I left myself a note before PTO that said something like "sigh... out->quiet causing headaches again"

If we set quiet in main, then we lose the other `out->info()` calls unless verbose is given. That includes the instruction messages and the "we're doing/just did <thing>" messages (I'm less tied to the latter category, but still).

I'm open to adding a "verbose" or "show_headers" argument to the shadow message and the helper functions. We've usually declined to add such arguments in the past because they're (effectively) specific to the text format. We've used approaches similar to this one with quiet.

Also open to other ideas

Reply:
It's fine as is